File: backend/app/schemas/agent_config.py
from typing import Literal

from pydantic import Field

from app.schemas.common import CamelModel

QualifyingField = Literal["budget", "timeline", "need", "companySize", "authority"]


class QualifyingQuestion(CamelModel):
    id: str
    field: QualifyingField
    prompt: str
    required: bool


GmailReplyMode = Literal["auto_send", "review_first"]


class AgentConfig(CamelModel):
    persona: str
    qualifying_questions: list[QualifyingQuestion] = Field(min_length=1, max_length=8)
    handoff_threshold: int = Field(ge=0, le=100)
    calendly_url: str | None = None
    guardrails: list[str]
    active: bool
    gmail_reply_mode: GmailReplyMode = "auto_send"


class AgentConfigUpdate(CamelModel):
    persona: str
    qualifying_questions: list[QualifyingQuestion] = Field(min_length=1, max_length=8)
    handoff_threshold: int = Field(ge=0, le=100)
    calendly_url: str | None = None
    guardrails: list[str]
    active: bool
    gmail_reply_mode: GmailReplyMode = "auto_send"

File: frontend/lib/schema/agent-config.ts
import { z } from "zod";

export const QualifyingQuestionSchema = z.object({
  id: z.string(),
  field: z.enum(["budget", "timeline", "need", "companySize", "authority"]),
  prompt: z.string(),
  required: z.boolean(),
});
export type QualifyingQuestion = z.infer<typeof QualifyingQuestionSchema>;

export const AgentConfigSchema = z.object({
  persona: z.string(),
  qualifyingQuestions: z.array(QualifyingQuestionSchema).min(1).max(8),
  handoffThreshold: z.number().min(0).max(100),
  calendlyUrl: z.string().nullable(),
  guardrails: z.array(z.string()),
  active: z.boolean(),
  // "auto_send" (default) or "review_first" — set once during onboarding Step 1,
  // editable later from Dashboard → Integrations. Gmail-specific: see
  // backend/app/jobs/gmail_poll.py.
  gmailReplyMode: z.enum(["auto_send", "review_first"]),
});
export type AgentConfig = z.infer<typeof AgentConfigSchema>;
